chainer/links/model/vision/vgg.py

```diff
@@ -0,0 +1,305 @@
+            ('fc8', [self.fc8, relu]),
+            ('prob', [softmax]),
+        ])
+
+    @property
+    def available_layers(self):
+        return list(self.functions.keys())
+
+    @classmethod
+    def convert_caffemodel_to_npz(cls, path_caffemodel, path_npz):
+        """Converts a pre-trained caffemodel to a chainer model.
+
+        Args:
+            path_caffemodel (str): Path of the pre-trained caffemodel.
+            path_npz (str): Path of the converted chainer model.
+        """
+
+        # As CaffeFunction uses shortcut symbols,
+        # we import CaffeFunction here.
+        from chainer.links.caffe.caffe_function import CaffeFunction
+        caffemodel = CaffeFunction(path_caffemodel)
+        npz.save_npz(path_npz, caffemodel, compression=False)
+
+    def __call__(self, x, layers=['prob'], test=True):
+        """Computes all the feature maps specified by ``layers``.
+
+        Args:
+            x (~chainer.Variable): Input variable.
+            layers (list of str): The list of layernames you want to extract.
+            test (bool): If ``True``, it runs in test mode.
+
+        Returns:
+            Dictionary of ~chainer.Variable: The directory in which
+            the key contains the layer name and the value contains
+            the corresponding variable.
+
+        """
+
+        h = x
+        activations = {}
+        target_layers = set(layers)
+        for key, funcs in self.functions.items():
+            if len(target_layers) == 0:
+                break
+            for func in funcs:
+                if func is dropout:
+                    h = func(h, train=not test)
+                else:
+                    h = func(h)
+            if key in target_layers:
+                activations[key] = h
+                target_layers.remove(key)
+        return activations
+
+    def extract(self, images, layers=['fc7'], size=(224, 224), test=True):
+        """Extracts all the feature maps of given images.
+
+        The difference of directly executing ``__call__`` is that
+        it directly accepts images as an input and automatically
+        transforms them to a proper variable. That is,
+        it is also interpreted as a shortcut method that implicitly call
+        ``prepare`` and ``__call__`` functions.
+
+        Args:
+            image (iterable of PIL.Image or numpy.ndarray): Input images.
+            layers (list of str): The list of layernames you want to extract.
+            size (pair of ints): The resolution of resized images used as
+                an input of CNN. All the given images are not resized
+                if this argument is ``None``, but the resolutions of
+                all the images should be the same.
+            test (bool): If ``True``, it runs in test mode.
+
+        Returns:
+            Dictionary of ~chainer.Variable: The directory in which
+            the key contains the layer name and the value contains
+            the corresponding variable.
```

Comment:
ditto
Maybe for clarity: 
corresponding variable -> corresponding feature map variable